File: reto_clean_code/python/step4_informe_original.py
"""
Ejemplo de una clase que viola el Principio de Responsabilidad Única al
mezclar generación, notificación y persistencia en una sola entidad.
Utilízala como base para refactorizar en el paso 4.
"""


class Informe:
    def generar_pdf(self) -> None:
        # Generar PDF
        pass

    def enviar_email(self) -> None:
        # Enviar correo
        pass

    def guardar_en_bd(self) -> None:
        # Guardar en base de datos
        pass

File: reto_clean_code/angular/step4-informe-original.ts
// Clase que viola el Principio de Responsabilidad Única al combinar
// generación, notificación y persistencia en un único servicio.  Utiliza
// este archivo para realizar la refactorización en el paso 4.

export class Informe {
  generarPdf(): void {
    // Generar PDF
  }

  enviarEmail(): void {
    // Enviar por correo electrónico
  }

  guardarEnBaseDatos(): void {
    // Guardar en la base de datos
  }
}